ocaml/src/asd_server.ml

```diff
@@ -682,6 +682,87 @@ let execute_query : type req res.
     | Capabilities ->
        fun () ->
        return' capabilities
+    | RangeValidate ->
+       fun ({ RangeQueryArgs.first;finc;last;reverse;max;},
+            verify_checksum, show_all, prio) ->
+       let exists_ok (checksum,blob) =
+         match blob with
+           | Value.Direct v  -> Lwt.return None
+           | Value.OnFs(fnr,size) ->
+              let file_name = DirectoryInfo.get_file_path dir_info fnr in
+              Lwt_unix.file_exists file_name >>= fun ok ->
+              Some ("file_exists",ok) |> Lwt.return
+       in
+       let checksum_ok cv =
+         let (checksum, _) = cv in
+         let algo = Checksum.algo_of checksum in
+         Lwt.catch
+           (fun () ->
+             Value.get_blob_from_value dir_info cv >>= fun blob ->
+             let hash = Hashes.make_hash algo in
+             let open Slice in
+             let () = hash # update_substring blob.buf blob.offset blob.length in
```

Comment:
I think we want to calculate the hashes of big blobs on other threads...
So files should be read into a Lwt_bytes.t (avoid reading to Bytes.t and then make another copy...) and then use `hash # update_lwt_bytes_detached`